CoinGecko.cs

```diff
@@ -92,65 +102,60 @@ public override BaseData Reader(SubscriptionDataConfig config, string line, Date
         /// <returns>A clone of the object</returns>
         public override BaseData Clone()
         {
-            return new MyCustomDataType
+            return new CoinGecko
             {
                 Symbol = Symbol,
                 Time = Time,
                 EndTime = EndTime,
-                SomeCustomProperty = SomeCustomProperty,
+                MarketCap = MarketCap,
+                Value = Value,
+                Volume = Volume
             };
         }
 
         /// <summary>
-        /// Indicates whether the data source is tied to an underlying symbol and requires that corporate events be applied to it as well, such as renames and delistings
+        /// Creates a Symbol object for a given market and quote currency
         /// </summary>
-        /// <returns>false</returns>
-        public override bool RequiresMapping()
+        /// <param name="market">The market the ticker resides in</param>
+        /// <param name="quoteCurrency">The quote currency of the crypto-currency pair. E.g. USD for BTCUSD</param>
+        /// <returns>A new Symbol object for the specified ticker</returns>
+        public Symbol CreateSymbol(string market, string quoteCurrency = "USD")
         {
-            return true;
+            return Symbol.Create($"{Coin}{quoteCurrency}", SecurityType.Crypto, market);
```

Comment:
Think `SecurityType` should probably be an optional argument too, thinking about the crypto futures potential use case here